samtranslator/utils/traverse.py

```diff
@@ -0,0 +1,142 @@
+from typing import Any, Callable, Dict, List
+
+
+class Traverse:
+    """
+    Base Class for traversing
+    """
+
+    def traverse_wrapper(self, input_value: Any, resolution_data: Dict[str, Any], resolver_method_name: str) -> Any:
+        """
+        Wraps the Driver Method which allows the resolver method to be changed as needed
+
+        :param input_value: Any primitive type  (dict, array, string etc) whose value might contain a changed value
+        :param resolution_data: Data that will help with resolution. For example, changed logical ids from hashing
+        LayerVersion
+        :param resolver_method_name: Name of the method that will be called to actually resolve an the function. This method
+            is called with the parameters `(input, resolution_data)`.
+        :return: Modified `input` with values resolved
+        """
+        if resolver_method_name == "DependsOn":
+            return self._traverse(input_value, resolution_data, self._resolve_depends_on)
+
+        return input_value
+
+    # here is where I would put constant variables
+    def _traverse(
```

Comment:
These functions are same as IntrinsicResolver. Would be good to share these functions between 2 classes.